pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -28,59 +24,17 @@ const (
 	ProjectTagCount             int = 1
 )
 
-type SecurityPolicyService struct {
+type Service struct {
+	common.Service
 	Client              client.Client
-	NSXClient           *nsx.Client
-	NSXConfig           *config.NSXOperatorConfig
 	GroupStore          cache.Indexer
 	SecurityPolicyStore cache.Indexer
 	RuleStore           cache.Indexer
 }
 
-var log = logf.Log.WithName("service").WithName("firewall")
-
-// InitializeSecurityPolicy sync NSX resources
-func InitializeSecurityPolicy(NSXClient *nsx.Client, cf *config.NSXOperatorConfig) (*SecurityPolicyService, error) {
-	wg := sync.WaitGroup{}
-	wgDone := make(chan bool)
-	fatalErrors := make(chan error)
-
-	wg.Add(3)
-	service := &SecurityPolicyService{NSXClient: NSXClient}
-	service.GroupStore = cache.NewIndexer(keyFunc, cache.Indexers{
-		util.TagScopeNamespace:           namespaceIndexFunc,
-		util.TagScopeSecurityPolicyCRUID: securityPolicyCRUIDScopeIndexFunc,
-		util.TagScopeRuleID:              ruleCRUIDScopeIndexFunc,
-	})
-	service.SecurityPolicyStore = cache.NewIndexer(keyFunc, cache.Indexers{
-		util.TagScopeSecurityPolicyCRUID: securityPolicyCRUIDScopeIndexFunc,
-	})
-	service.RuleStore = cache.NewIndexer(keyFunc, cache.Indexers{
-		util.TagScopeSecurityPolicyCRUID: securityPolicyCRUIDScopeIndexFunc,
-	})
-	service.NSXConfig = cf
-
-	go queryGroup(service, &wg, fatalErrors)
-	go querySecurityPolicy(service, &wg, fatalErrors)
-	go queryRule(service, &wg, fatalErrors)
-
-	go func() {
-		wg.Wait()
-		close(wgDone)
-	}()
-
-	select {
-	case <-wgDone:
-		break
-	case err := <-fatalErrors:
-		close(fatalErrors)
-		return service, err
-	}
-
-	return service, nil
-}
+var log = common.Log
 
-func (service *SecurityPolicyService) OperateSecurityPolicy(obj *v1alpha1.SecurityPolicy) error {
+func (service *Service) OperateSecurityPolicy(obj *v1alpha1.SecurityPolicy) error {
```

Comment:
Please change Service back to SecurityPolicyService

Reply:
ok